### PR TITLE
[MOB-2543] Personal Counter Removal

### PR DESCRIPTION
**Pull request by d4r1091**
[MOB-2543]

## Context

As part of the Guild days, we picked the Personal Counter Removal project intending to remove all the dependencies made out of the personal counter feature.

## Other

- Left the logic inherent to showing the Push Notification Consent screen as part of the codebase, passing a temporary `0` for SearchCount.  Created a follow-up [ticket](https://ecosia.atlassian.net/browse/MOB-2698)
- No need to point to the same Personal Counter Removal work on the `Core` side as this ticket removes all its dependencies and has been worked In conjunction with the `Core`'s work (pointing to its local) to ensure all dependencies have been cleaned up. 

### Checklist

- [x] I performed some relevant testing on a real device and/or simulator
- [x] I added the `// Ecosia:` helper comments where needed
- [x] I am using the correct version from [core](https://github.com/ecosia/ios-core), including any necessary changes for this ticket and pointing to `main` instead of a feature branch

[MOB-2543]: https://ecosia.atlassian.net/browse/MOB-2543?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Client/TabManagement/Legacy/LegacyTabManager.swift [hunk 1]
```diff
@@ -179,13 +179,6 @@ class LegacyTabManager: NSObject, FeatureFlaggable, TabManager, TabEventHandler
                                                object: nil)
         // Ecosia: Cookie and settings observing
         configuration.websiteDataStore.httpCookieStore.add(self)
-
-        searchSettingsObserver = NotificationCenter.default
-            .publisher(for: .searchSettingsChanged)
-            .sink() { [privateConfiguration, configuration] _ in
-                configuration.websiteDataStore.httpCookieStore.setCookie(Cookie.makeStandardCookie())
-                privateConfiguration.websiteDataStore.httpCookieStore.setCookie(Cookie.makeIncognitoCookie())
-            }
```
Comment (lucaschifino) [inline Client/TabManagement/Legacy/LegacyTabManager.swift:188]: I guess this is still relevant for keeping our search settings updated with web, why were they removed?

**Review by lucaschifino — COMMENTED** [review]
Added some considerations, let me know what you think and if I'm missing some context for the changes 🙏

File: Client/Ecosia/Analytics/Analytics.swift [hunk 2]
```diff
@@ -156,16 +156,10 @@ final class Analytics {
...
-        // we add the number of search counts as value of the event
-        if action == .view {
-            event.value = NSNumber(integerLiteral: User.shared.searchCount)
-        }
-        
         // Add context (if any) from current EngagementService enabled
         if let toggleName = Unleash.Toggle.Name(rawValue: EngagementServiceExperiment.toggleName),
            let context = Self.getTestContext(from: toggleName) {
-            event.contexts.append(context)
+            event.entities.append(context)
```
[the remark this answers is not in the document]
Comment (d4r1091) [reply Client/Ecosia/Analytics/Analytics.swift:162]: Regarding `event.entities.append(context)`, it's been renamed only. Nothing else 👍

File: Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift [hunk 3]
```diff
@@ -189,8 +189,7 @@ class BrowserViewController: UIViewController,
     }
 
     fileprivate var shouldShowDefaultBrowserPromo: Bool {
-        profile.prefs.intForKey(PrefsKeys.IntroSeen) == nil &&
-        DefaultBrowserExperiment.minPromoSearches() <= User.shared.searchCount
```
[the remark this answers is not in the document]
Comment (d4r1091) [reply Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:193]: The `DefaultBrowserExperiment` is still enabled but solely related to the stored `PrefsKeys.IntroSeen`

**Comment by d4r1091** [comment]
@lucaschifino thanks for your comments between here and `Core` I remembered your question on All-Hands so I could re-evaluate this from a business perspective and not a user perspective only and realized that we could indeed keep the `searchCount` alongside its logic overall implementation as long as we don't show it at the UI level. I've reviewed the overall PR 💪. Even this one is not needed https://github.com/ecosia/ios-browser/pull/714. Let's say I do good brain-refresh exercises in case of certain parts of our codebase. 😅

File: Client/Ecosia/Extensions/AppSettingsTableViewController+Ecosia.swift [hunk 4]
```diff
@@ -17,12 +17,6 @@ extension AppSettingsTableViewController {
             getEcosiaSupportSection(),
             getEcosiaAboutSection()
         ]
-        
-        // Ecosia: Not adding search settings with Bing distribution experiment
-        if !BingDistributionExperiment.isEnabled {
-            sections.insert(getSearchSection(), at: 1)
-        }
```
Comment (lucaschifino) [inline Client/Ecosia/Extensions/AppSettingsTableViewController+Ecosia.swift:24]: What is the reason to remove this? I guess given `BingDistributionExperiment` still exists it still makes sense to remove Ecosia search settings whenever it's enabled.

File: Client/Ecosia/Extensions/AppSettingsTableViewController+Ecosia.swift [hunk 5]
```diff
@@ -153,8 +147,6 @@ extension AppSettingsTableViewController {
             CreateReferralCode(settings: self),
             AddReferral(settings: self),
             AddClaim(settings: self),
-            ChangeSearchCount(settings: self),
-            ResetSearchCount(settings: self),
```
Comment (lucaschifino) [inline Client/Ecosia/Extensions/AppSettingsTableViewController+Ecosia.swift:157]: Given we still use the search count internally for experiments, I guess this debug settings are still useful for testing. What do you think about us keeping them?

File: Client/Ecosia/UI/NTP/Impact/ClimateImpactInfo.swift [hunk 6]
```diff
@@ -112,21 +95,10 @@ enum ClimateImpactInfo: Equatable {
...
-            return nil
-        }
-    }
-    
+        
     /// Created to be used for comparison without taking the associated types arguments into consideration.
     var rawValue: Int {
         switch self {
-        case .search:
-            return 0
```
Comment (lucaschifino) [inline Client/Ecosia/UI/NTP/Impact/ClimateImpactInfo.swift:129]: Given this is used for comparison, shouldn't we update the others so it starts on `0`? I guess this is minor and not a blocker, even wondering if there is a better way to do it 🤔

File: Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (lucaschifino) [reply Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:193]: Given we keep `User.shared.searchCount`, why not continue comparing it here? I guess it still might be relevant for the experiment

File: Client/Frontend/Browser/SearchEngines/OpenSearchEngine.swift [hunk 7]
```diff
@@ -93,8 +93,6 @@ class OpenSearchEngine: NSObject, NSSecureCoding {
            let fromTemplateURL = getURLFromTemplate(searchTemplate, query: query),
            fromTemplateURL.baseDomain != "ecosia.org" {
             return fromTemplateURL
-        } else if BingDistributionExperiment.isEnabled {
-            return BingDistributionExperiment.searchURLForQuery(query)
```
Comment (lucaschifino) [inline Client/Frontend/Browser/SearchEngines/OpenSearchEngine.swift:97]: Given `BingDistributionExperiment` still exists, I'd also argue this is needed if it's enable given this is the main logic for the experiment.

**Review by lucaschifino — COMMENTED** [review]
Thank you for the adjustments ❤️ 

Nothing major, but added some comments specially with regards to keeping some experiments still properly working if they still exist in code.

**Comment by d4r1091** [comment]
@lucaschifino the BingDistrition experiment is not running anymore.
Even though this ticket shouldn't be part of the Bing Distribution Removal work, I thought about Including it at least from the places where this check is done.
I'm happy to revert it and work it out separately.

**Review by lucaschifino — APPROVED** [review]
Looking great! 🎉 Thank you for the adjustments ❤️ 

> @lucaschifino the BingDistrition experiment is not running anymore.
Even though this ticket shouldn't be part of the Bing Distribution Removal work, I thought about Including it at least from the places where this check is done.
I'm happy to revert it and work it out separately.

I believe that is better so that we either leave the experiment in a working state (even if not used for now) or we remove it completely.

**Comment by d4r1091** [comment]
@lucaschifino thanks for the back to back!

**Pull request merged**